mrp_bom_component_cost/tests/test_mrp_bom_component_cost.py

```diff
@@ -0,0 +1,102 @@
+        cls.component2 = cls.env["product.product"].create(
+            {
+                "name": "Test Component 2",
+                "detailed_type": "product",
+                "standard_price": 12,
+                "uom_id": cls.env.ref("uom.product_uom_dozen").id,
+            }
+        )
+        cls.component_eur = cls.env["product.product"].create(
+            {
+                "name": "Test Component Eur",
+                "detailed_type": "product",
+                "standard_price": 1,
+                "company_id": cls.company_eur.id,
+            }
+        )
+        cls.bom = cls.env["mrp.bom"].create(
+            {
+                "product_id": cls.product.id,
+                "product_tmpl_id": cls.product.product_tmpl_id.id,
+                "bom_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.component1.id,
+                            "product_qty": 2,
+                            "product_uom_id": cls.env.ref("uom.product_uom_dozen").id,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.component2.id,
+                            "product_qty": 1,
+                            "product_uom_id": cls.env.ref("uom.product_uom_unit").id,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    def test_mrp_bom_component_cost(self):
+        self.assertEqual(self.bom.bom_line_ids[0].cost_unit, 12)
+        self.assertEqual(self.bom.bom_line_ids[0].cost_subtotal, 24)
+        self.assertEqual(self.bom.cost_subtotal, 25)
+
+    def test_mrp_bom_component_cost_multicurrency(self):
+        self.bom = self.env["mrp.bom"].create(
+            {
+                "product_id": self.product.id,
+                "product_tmpl_id": self.product.product_tmpl_id.id,
+                "company_id": self.company_eur.id,
+                "bom_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.component1.id,
+                            "product_qty": 1,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.component_eur.id,
+                            "product_qty": 1,
+                        },
+                    ),
+                ],
+            }
+        )
+        line_usd = self.bom.bom_line_ids[0]
+        self.assertNotEqual(self.bom.currency_id, line_usd.product_id.currency_id)
+        self.assertNotEqual(line_usd.cost_unit, 1)
+        self.assertNotEqual(line_usd.cost_subtotal, 1)
+        self.assertNotEqual(self.bom.cost_subtotal, 2)
```

Comment:
Add a new test that checks for changes in product prices once production orders have been generated. What happens if the pump is changed? Is everything recalculated?

Reply:
The new fields are just informative. This PR does not edit production orders. If a product price changes, the change will be propagated to the bom price, but nothing more.

P.S.:: What do you mean with pump?

Reply:
Sorry for the term "pump," I wanted to indicate the list of materials in the production order.

Reply:
Then i don't think a new test is needed.

If a production order is generated nothing will change, and if the list of materials in a production order change nothing will happen

At most, i can edit the main test to edit a value of the product and see how the computed value changes

Reply:
And if they create it manually?
Let's use the case of a product with two production lists. What happens if I'm creating the order manually and can't create the first bill of materials because I don't have stock, so I switch to the second one? Has this behavior been reviewed?The values ​​must match the configured production list. What happens if I add or remove an extra product? What happens if I add steps to the production order?

Reply:
@manuelregidor Do you understand what Valentin requests? Dou you agree with him?

I have requested a meet with Valentin for more explanations, but he says I have to ask you before